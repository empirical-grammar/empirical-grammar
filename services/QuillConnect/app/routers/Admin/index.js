import request from 'request';

const usersEndpoint = `${process.env.EMPIRICAL_BASE_URL}/api/v1/users.json`;
const newSessionEndpoint = `${process.env.EMPIRICAL_BASE_URL}/session/new`;

export default {
  path: 'admin',
  indexRoute: { onEnter: (nextState, replace) => replace('/admin/question-health'), },
  getChildRoutes: (partialNextState, cb) => {
    fetch(usersEndpoint, {
        method: 'GET',
        mode: 'cors',
        credentials: 'include',
      }).then((response) => {
        if (!response.ok) {
          throw Error(response.statusText);
        }
        return response.json();
      }).then((response) => {
          if (response.user === null || (response.user.hasOwnProperty('role') && response.user.role !== 'staff')){
            window.location = newSessionEndpoint;
          }
          else {
            Promise.all([
              import(/* webpackChunkName: "admin-concept-feedback" */ './routes/ConceptFeedback/index.js'),
              import(/* webpackChunkName: "admin-concepts" */ './routes/Concepts/index.js'),
              import(/* webpackChunkName: "admin-dashboard" */ './routes/DataDash/index.js'),
              import(/* webpackChunkName: "admin-question-health" */ './routes/QuestionHealth/index.js'),
              import(/* webpackChunkName: "admin-title-cards" */ './routes/TitleCards/index.js'),
              import(/* webpackChunkName: "admin-lessons" */ './routes/Lessons/index.js'),
              import(/* webpackChunkName: "admin-questions" */ './routes/Questions/index.js'),
              import(/* webpackChunkName: "admin-fill-in-the-blanks" */ './routes/FillInTheBlanks/index.js'),
              import(/* webpackChunkName: "admin-sentence-fragments" */ './routes/SentenceFragments/index.js'),
              import(/* webpackChunkName: "admin-item-levels" */ './routes/ItemLevels/index.js')
            ])
            .then(modules => cb(null, modules.map(module => module.default)))
            // to do, use Sentry to capture error
            // .catch(err => console.error('Dynamic page loading failed', err));
          }
      }).catch((error) => {
        // to do, use Sentry to capture error
      })
  },
  getComponent(nextState, cb) {
    require.ensure([], (require) => {
      cb(null, require('components/admin/admin').default);
    }, 'admin-root');
  },
};
