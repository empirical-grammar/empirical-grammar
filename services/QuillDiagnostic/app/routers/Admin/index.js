import request from 'request';

const usersEndpoint = `${process.env.EMPIRICAL_BASE_URL}/api/v1/users.json`;
const newSessionEndpoint = `${process.env.EMPIRICAL_BASE_URL}/session/new`;

export default {
  path: 'admin',
  /*indexRoute: { onEnter: (nextState, replace) => replace('/admin/question-health'), },*/
  indexRoute: { onEnter: (nextState, replace) => (request(usersEndpoint, (error, response, body) => { if (!error && response.statusCode === 200) { const json_body = JSON.parse(body);if (json_body.user === null || (json_body.user.hasOwnProperty('role') && json_body.user.role !== 'staff')){ window.location = newSessionEndpoint } else { replace('/admin/question-health') }}}}),
  getChildRoutes: (partialNextState, cb) => {
    Promise.all([
      import(/* webpackChunkName: "admin-concept_feedback" */ './routes/ConceptFeedback/index.js'),
      import(/* webpackChunkName: "admin-concepts" */ './routes/Concepts/index.js'),
      import(/* webpackChunkName: "admin-dashboard" */ './routes/DataDash/index.js'),
      import(/* webpackChunkName: "admin-question-health" */ './routes/QuestionHealth/index.js'),
      import(/* webpackChunkName: "admin-title-cards" */ './routes/TitleCards/index.js'),
      import(/* webpackChunkName: "admin-lessons" */ './routes/Lessons/index.js'),
      import(/* webpackChunkName: "admin-questions" */ './routes/Questions/index.js'),
      import(/* webpackChunkName: "admin-fill-in-the-blanks" */ './routes/FillInTheBlanks/index.js'),
      import(/* webpackChunkName: "admin-sentence-fragments" */ './routes/SentenceFragments/index.js'),
      import(/* webpackChunkName: "admin-item-levels" */ './routes/ItemLevels/index.js'),
      import(/* webpackChunkName: "admin-item-levels" */ './routes/CloneConnectQuestions/index.js')
    ])
    .then(modules => cb(null, modules.map(module => module.default)))
    // to do, use Sentry to capture error
    // .catch(err => console.error('Dynamic page loading failed', err));
  },
  getComponent(nextState, cb) {
    require.ensure([], (require) => {
      cb(null, require('components/admin/admin').default);
    }, 'admin-root');
  },
};
