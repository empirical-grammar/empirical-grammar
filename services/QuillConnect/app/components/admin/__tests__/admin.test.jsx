import React from 'react';
import 'whatwg-fetch'
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Admin from "../admin"
import { Provider } from 'react-redux'
import createStore from '../../../utils/configureStore';
import AdminMainSidebar from '../adminMainSidebar.jsx'
import AdminLessonSidebar from '../adminLessonSidebar';
import Question from '../../questions/question'

describe('adminContainer component', () => {

  const store = createStore();

  it('should render with lessons sidebar when url contains lesson-view', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[ '/admin/lesson-view/lessonID/questions/questionID' ]}>
          <Admin />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(AdminLessonSidebar)).toHaveLength(1);
    expect(wrapper.find(Question)).toHaveLength(1);
  });

  it('should render with main sidebar when url doesn not contain lesson-view', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[ '/admin/lessons' ]}>
          <Admin />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(AdminMainSidebar)).toHaveLength(1);
  });
})
