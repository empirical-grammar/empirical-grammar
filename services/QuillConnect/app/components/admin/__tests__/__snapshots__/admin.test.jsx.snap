// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adminContainer component should render with lessons sidebar when url contains lesson-view 1`] = `ReactWrapper {}`;

exports[`adminContainer component should render with main sidebar when url doesn not contain lesson-view 1`] = `ReactWrapper {}`;
